### DISCUSSION TITLE
does every one using this ide? eclipse for arduino?

### DISCUSSION THREAD

**Discussion by youxiaojie** [dbuezas/lgt8fx#136] [General]
http://eclipse.baeyens.it/index.shtml

**Comment by SuperUserNameMan** [comment]
I did not know about this one.

I tried [UECIDE](https://github.com/UECIDE/UECIDE) a while ago. (I will have to try it again. EDIT : i tried it again. They support "Larduino HSP LGT8F Core", but I don't know how to add our "lgt8fx core" ).

Currently, I use Geany as external-editor, and Arduino IDE as compiler and uploader.

**Reply by youxiaojie** [reply]
I hope you use context function. this is the only ide I can found.and support our package. support native uploader.

**Comment by LaZsolt** [comment]
@youxiaojie 
I didn't know about it either. But if I'll have a lot of free time, I'll try it.
Andreas Spiess recomennded an another tool (Visual Micro) (Windows only)) : https://www.youtube.com/watch?v=7SiprW-goj0
And, the newer version of Arduino evinronment: https://www.arduino.cc/pro/arduino-pro-ide

**Comment by LaZsolt** [comment]
New Arduino IDE released. https://www.youtube.com/watch?v=nlI_5vxm3bk

**Reply by SuperUserNameMan** [reply]
https://github.com/arduino/arduino-ide/releases

**Comment by prosper00** [comment]
I've mostly switched to using arduino-cli (https://www.arduino.cc/pro/cli). I use Geany or VSCode as the editor, and I wrote a crude Makefile to invoke the build (https://github.com/prosper00/arpeggiator/blob/master/Makefile)

**Comment by dbuezas** [comment]
Am I the only one using the Arduino vscode extension?

**Reply by prosper00** [reply]
I've tried it. Definitely a big step up from Arduino IDE. We'll see how it stacks up against the new IDE (which I believe is also electron-based, like VSCode.)

In general though, I'm not a huge fan of IDE's. There's just too much hand-wavy behind-the-scenes stuff that's supposed to 'just work,' but invariably needs to be tweaked and tuned to get it to work the way I want it to. And because it's hidden behind the scenes, it's more difficult to tweak and I find that I spend more time trying to find documentation for and fiddling with my IDE than I do *using* my IDE. 

At least with arduino-cli, I can separate out the 'editor' bits from the 'build' bits, so that I can use whichever editor I want [vim, geany, vscode, or even Arduino GUI], and script the build bits (via make) and get the exact same results. I have seen the VSCode plugin (and the platformio Arduino framework) produce different binaries. Probably comes down to different compiler versions or something innocuous, but, it bugs me. I hate debugging my environment.
